economy/schema.py

```diff
@@ -159,6 +187,51 @@ def resolve_my_bank_account(self, info, *args, **kwargs):
     def resolve_all_soci_bank_accounts(self, info, *args, **kwargs):
         return SociBankAccount.objects.all()
 
+    def resolve_my_expenditures(self, info, date_range, *args, **kwargs):
+        data = []
+        total = 0
+
+        my_bank_account = info.context.user.bank_account
+
+        cal = calendar.Calendar()
+        today = datetime.date.today()
+        iterator = cal.itermonthdates(today.year, today.month)
+        for day in iterator:
+            # Won't be any more data in the future
+            if day > today:
+                data.append(ExpenditureDay(day=day, sum=0))
+                continue
+
+            # The iterator returns days in prev month if they are contained in within the first week
+            if day.month != today.month:
+                continue
+
+            # Create a day datetime range for each day
+            day_min = timezone.make_aware(
+                timezone.datetime(
+                    year=day.year,
+                    month=day.month,
+                    day=day.day,
+                    hour=0,
+                    minute=0,
+                    second=0,
+                )
+            )
+            day_max = day_min + timezone.timedelta(hours=23, minutes=59, seconds=59)
+            # https://blog.zdsmith.com/posts/comparing-dates-and-datetimes-in-the-django-orm.html
+
+            purchases = my_bank_account.product_orders.filter(
+                purchased_at__range=(day_min, day_max)
+            )
```

Comment:
Could probably optimize this by querying purchases for the relevant month outside the loop and then we filter that QS for a given day range inside the loop instead